test/Tests/src/Semantic/Warnings_Spec.enso

```diff
@@ -306,7 +306,7 @@ spec = Test.group "Dataflow Warnings" <|
         res = [1, 2, 3, 4].map nested_range_even
 
         res . should_equal [[0], [0, 1], [0, 1, 0], [0, 1, 0, 1]]
-        Warning.get_all res . map .value . should_equal [3, 2, 1, 0, 2, 1, 0, 1, 0, 0]
+        Warning.get_all_wrapped res . map .value . should_equal [(Map_Error.Error 3 (Map_Error.Error 3 3)), (Map_Error.Error 3 (Map_Error.Error 2 2)), (Map_Error.Error 3 (Map_Error.Error 1 1)), (Map_Error.Error 3 (Map_Error.Error 0 0)), (Map_Error.Error 2 (Map_Error.Error 2 2)), (Map_Error.Error 2 (Map_Error.Error 1 1)), (Map_Error.Error 2 (Map_Error.Error 0 0)), (Map_Error.Error 1 (Map_Error.Error 1 1)), (Map_Error.Error 1 (Map_Error.Error 0 0)), (Map_Error.Error 0 (Map_Error.Error 0 0))]
```

Comment:
Why remove the first test? Shall we test _both_ `get_all_wrapped` as well as `get_all`?

Also, could you explain a bit the design here? I assume it's been discussed with @hubertp but I was a bit out of the loop about it and I'd like to understand it better.

When are we supposed to rely on `Warning.get_all` and when on `Warning.get_all_wrapped`?

Reply:
Also ideally I think the idea for the design should be documented _somewhere_ so that we can refer to it in the future.

Reply:
Yes, I believe the conclusion was that we will display the index of the deeply nested warning by nesting the warning. Might be worth documenting that it is from outer to inner array/ I think the decision took place as part of the ticket discussion and meetings.

Reply:
Sounds good, I think it would be good to have some high level overview.

Reply:
I added some notes in the PR description.

Reply:
Restored the removed test.
Added documentation of nested wrappers.
Also added detailed documentation in `docs/`.